erpnext/selling/page/point_of_sale/point_of_sale.js

```diff
@@ -714,6 +715,21 @@ class POSCart {
 		this.customer_field.set_value(this.frm.doc.customer);
 	}
 
+  	disable_numpad_control() {
+		if(!this.frm.allow_edit_rate && !this.frm.allow_edit_discount) {
+			return ['Rate', 'Disc'];
+		}
+		if(!this.frm.allow_edit_rate || !this.frm.allow_edit_discount) {
+			if(!this.frm.allow_edit_rate) {
+				return ['Rate'];
+			} else {
+				return ['Disc'];
+			}
+		} else {
+			return [];
+		}
+	}
+
```

Comment:
@jay-parikh 

You can also write above function as below
```
disable_numpad_control() { 
	let disabled_btns = [];

	if(!this.frm.allow_edit_rate) {
		disabled_btns.push('Rate')
	}

	if(!this.frm.allow_edit_discount) {
		disabled_btns.push('Disc')
	}

	return disabled_btns;
}
```

Reply:
@rohitwaghchaure Thanks for feedback. Let me rewrite and check.